src/models/blockchain/block/block_body.rs

```diff
@@ -78,6 +127,25 @@ impl MastHash for BlockBody {
             self.block_mmr_accumulator.encode(),
         ]
     }
+
+    fn merkle_tree(&self) -> MerkleTree {
+        self.merkle_tree
+            .get_or_init(|| {
+                let mut digests = self
```

Comment:
ideally we would just call the default impl inside the get_or_init(), but sadly rust does not permit that presently.

There is an RFC for it though, and I just added a comment:
https://github.com/rust-lang/rfcs/pull/3329#issuecomment-2513072126